src/queue.ts

```diff
@@ -0,0 +1,1052 @@
+                ({ MessageBody: Message, ...entry }) => ({
+                  ...entry,
+                  MessageBody: serialize(Message),
+                })
+              ),
+              QueueUrl: updatedQueueUrl,
+            })
+            .promise();
+
+          return response;
+        },
+      },
+      asl: (call, context) => {
+        const entries = call.args[0]?.expr;
+        if (entries === undefined) {
+          throw new SynthError(
+            ErrorCodes.Invalid_Input,
+            `the first argument 'input' is required by sendMessageBatch`
+          );
+        }
+
+        this.resource.grantSendMessages(context.role);
+
+        if (this.serializer.dataType === DataType.String) {
+          // when the data type is a String, no extra processing is required
+          // so we evaluate the entries to a JsonPath or Literal and pass it straight through
+          return context.evalExprToJsonPathOrLiteral(entries, (entries) => {
+            if (
+              ASLGraph.isJsonPath(entries) ||
+              (entries.value !== null && typeof entries.value === "object")
+            ) {
+              return context.stateWithHeapOutput({
+                Type: "Task",
+                Resource: "arn:aws:states:::aws-sdk:sqs:sendMessageBatch",
+                Parameters: {
+                  QueueUrl: this.queueUrl,
+                  ...(ASLGraph.isJsonPath(entries)
+                    ? {
+                        "Entries.$": `${entries.jsonPath}.Entries`,
+                      }
+                    : (entries.value as object)),
+                },
+                Next: ASLGraph.DeferNext,
+              });
+            } else {
+              throw new SynthError(
+                ErrorCodes.Invalid_Input,
+                `invalid 'Entries' property in SendMessageBatchRequest`
+              );
+            }
+          });
+        } else if (this.serializer.dataType === DataType.Json) {
+          // when the data type is Json, map over each of the entries and serialize them to JSON
+          return context.evalExprToJsonPath(
+            entries,
+            (entries, { addState }) => {
+              const heapVar = context.newHeapVariable();
+              addState({
+                Type: "Map",
+                ItemsPath: `${entries.jsonPath}.Entries`,
+                Parameters: {
+                  "entry.$": "$$.Map.Item.Value",
+                },
+                ResultPath: heapVar,
+                Next: ASLGraph.DeferNext,
+                Iterator: {
+                  StartAt: "serialize Message",
+                  States: {
+                    "serialize Message": {
+                      Type: "Pass",
+                      Parameters: {
+                        "value.$":
+                          this.serializer.dataType === DataType.Json
+                            ? `States.JsonToString($.entry.MessageBody)`
+                            : "$.entry.MessageBody",
```

Comment:
can you skip this if the map will be a no-op?

Reply:
Oh it is, I just refactored the code into two branches. This ternary is no longer needed.